driver/take_driver.hpp

```diff
@@ -0,0 +1,396 @@
+int TakeDriver<Tgpu, Tref>::GetandSetData()
+{
+    std::vector<int> in_len = GetInputTensorLengthsFromCmdLine();
+    SetTensorNd(inputDesc, in_len, data_type);
+    std::vector<int> out_len = GetOutputTensorLengthsFromCmdLine();
+    SetTensorNd(outputDesc, out_len, data_type);
+    SetTensorNd(indexDesc, out_len, data_type);
+
+    return 0;
+}
+
+template <typename Tgpu, typename Tref>
+std::vector<int> TakeDriver<Tgpu, Tref>::GetInputTensorLengthsFromCmdLine()
+{
+    int in_n = inflags.GetValueInt("in_n");
+    int in_c = inflags.GetValueInt("in_c");
+    int in_w = inflags.GetValueInt("in_w");
+    int in_h = inflags.GetValueInt("in_h");
+    int in_d = inflags.GetValueInt("in_d");
+
+    if((in_n != 0) && (in_c != 0) && (in_d != 0) && (in_h != 0) && (in_w != 0))
+    {
+        return std::vector<int>({in_n, in_c, in_d, in_h, in_w});
+    }
+    else if((in_n != 0) && (in_c != 0) && (in_h != 0) && (in_w != 0))
+    {
+        return std::vector<int>({in_n, in_c, in_h, in_w});
+    }
+    else if((in_n != 0) && (in_c != 0) && (in_w != 0))
+    {
+        return std::vector<int>({in_n, in_c, in_w});
+    }
+    else if((in_n != 0) && (in_w != 0))
+    {
+        return std::vector<int>({in_n, in_w});
+    }
+    else if(in_n != 0)
+    {
+        return std::vector<int>({in_n});
+    }
+    else
+    {
+        std::cerr << "Error Input Tensor Lengths\n" << std::endl;
+        return std::vector<int>({0});
+    }
+}
+
+template <typename Tgpu, typename Tref>
+std::vector<int> TakeDriver<Tgpu, Tref>::GetOutputTensorLengthsFromCmdLine()
+{
+    int out_n = inflags.GetValueInt("out_n");
+    int out_c = inflags.GetValueInt("out_c");
+    int out_w = inflags.GetValueInt("out_w");
+    int out_h = inflags.GetValueInt("out_h");
+    int out_d = inflags.GetValueInt("out_d");
+
+    if((out_n != 0) && (out_c != 0) && (out_d != 0) && (out_h != 0) && (out_w != 0))
+    {
+        return std::vector<int>({out_n, out_c, out_d, out_h, out_w});
+    }
+    else if((out_n != 0) && (out_c != 0) && (out_h != 0) && (out_w != 0))
+    {
+        return std::vector<int>({out_n, out_c, out_h, out_w});
+    }
+    else if((out_n != 0) && (out_c != 0) && (out_w != 0))
+    {
+        return std::vector<int>({out_n, out_c, out_w});
+    }
+    else if((out_n != 0) && (out_w != 0))
+    {
+        return std::vector<int>({out_n, out_w});
+    }
+    else if(out_n != 0)
+    {
+        return std::vector<int>({out_n});
+    }
+    else
+    {
+        std::cerr << "Error Output Tensor Lengths\n" << std::endl;
+        return std::vector<int>({0});
+    }
+}
+
+template <typename Tgpu, typename Tref>
+int TakeDriver<Tgpu, Tref>::AllocateBuffersAndCopy()
+{
+    size_t in_sz = GetTensorSize(inputDesc);
+    // index_sz = out_sz
+    size_t out_sz = GetTensorSize(outputDesc);
+
+    uint32_t ctx = 0;
+
+    in_dev    = std::unique_ptr<GPUMem>(new GPUMem(ctx, in_sz, sizeof(Tgpu)));
+    index_dev = std::unique_ptr<GPUMem>(new GPUMem(ctx, out_sz, sizeof(Tgpu)));
```

Comment:
```
index_dev = std::unique_ptr<GPUMem>(new GPUMem(ctx, out_sz, sizeof(Tgpu)));
index = std::vector<int32_t>(out_sz, static_cast<Tgpu>(0));
index[i] = prng::gen_A_to_B<int32_t>(static_cast<Tgpu>(-10), static_cast<Tgpu>(10));
...
```
The index tensor has a different dtype. Index should not be determined using Tgpu.

Reply:
[UPDATE] Fixed it in [this commit](https://github.com/moreh-dev/MIOpen/pull/9/commits/486f3b855803d176e03e747e830dcb27e69d44bb)